### PR TITLE
rauc: update build system to use meson

### PR DESCRIPTION
**Pull request by Emantor**
###### Description of changes

Adjust the build system to use meson instead of autotools.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 219935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/tools/misc/rauc/default.nix [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 , glib
 , json-glib
 , lib
 , nix-update-script
 , openssl
 , pkg-config
 , stdenv
+, meson
+, ninja
+, utillinux
```
Comment (symphorien) [inline pkgs/tools/misc/rauc/default.nix:14]:
```suggestion
, util-linux
```
this is an alias and should not be used in nixpkgs

Comment (SuperSandro2000) [reply pkgs/tools/misc/rauc/default.nix:14]: I am not sure why this is not triggering an error for ofborg

File: pkgs/tools/misc/rauc/default.nix [hunk 2]
```diff
@@ -22,21 +27,29 @@ stdenv.mkDerivation rec {
...
     updateScript = nix-update-script { };
   };
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ pkg-config autoreconfHook ];
+  nativeBuildInputs = [ pkg-config meson ninja ];
 
-  buildInputs = [ curl dbus glib json-glib openssl ];
+  buildInputs = [ curl dbus glib json-glib openssl utillinux libnl systemd ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/rauc/default.nix:45]: util-linux usually goes to nativeBuildInputs

File: pkgs/tools/misc/rauc/default.nix (hunk 1 above)
Comment (Emantor) [reply pkgs/tools/misc/rauc/default.nix]: Fixed.

File: pkgs/tools/misc/rauc/default.nix (hunk 2 above)
Comment (Emantor) [reply pkgs/tools/misc/rauc/default.nix]: `rauc` requires fdisk as a run-time dependency, so in this case `buildInputs` should be correct.

**Comment by Emantor** [comment]
@SuperSandro2000 comments should be resolved.

**Pull request merged**